runtime/compiler/codegen/J9AheadOfTimeCompile.cpp

```diff
@@ -324,6 +325,68 @@ J9::AheadOfTimeCompile::initializeCommonAOTRelocationHeader(TR::IteratedExternal
          }
          break;
 
+      case TR_InlinedStaticMethodWithNopGuard:
+      case TR_InlinedSpecialMethodWithNopGuard:
+      case TR_InlinedVirtualMethodWithNopGuard:
+      case TR_InlinedInterfaceMethodWithNopGuard:
+         {
+         TR_RelocationRecordNopGuard *inlinedMethod = reinterpret_cast<TR_RelocationRecordNopGuard *>(reloRecord);
+         uintptrj_t destinationAddress = reinterpret_cast<uintptrj_t>(relocation->getTargetAddress());
+         TR_VirtualGuard *guard = reinterpret_cast<TR_VirtualGuard *>(relocation->getTargetAddress2());
+
+         uint8_t flags = 0;
+         // Setup flags field with type of method that needs to be validated at relocation time
+         if (guard->getSymbolReference()->getSymbol()->getMethodSymbol()->isStatic())
+            flags = inlinedMethodIsStatic;
+         else if (guard->getSymbolReference()->getSymbol()->getMethodSymbol()->isSpecial())
+            flags = inlinedMethodIsSpecial;
+         else if (guard->getSymbolReference()->getSymbol()->getMethodSymbol()->isVirtual())
+            flags = inlinedMethodIsVirtual;
+         TR_ASSERT((flags & RELOCATION_CROSS_PLATFORM_FLAGS_MASK) == 0,  "reloFlags bits overlap cross-platform flags bits\n");
+
+         int32_t inlinedSiteIndex = guard->getCurrentInlinedSiteIndex();
+
+         TR_ResolvedMethod *resolvedMethod;
+         if (kind == TR_InlinedInterfaceMethodWithNopGuard)
+            {
+            TR_InlinedCallSite *inlinedCallSite = &comp->getInlinedCallSite(inlinedSiteIndex);
+            TR_AOTMethodInfo *aotMethodInfo = (TR_AOTMethodInfo *)inlinedCallSite->_methodInfo;
+            resolvedMethod = aotMethodInfo->resolvedMethod;
+            }
+         else
+            {
+            resolvedMethod = guard->getSymbolReference()->getSymbol()->getResolvedMethodSymbol()->getResolvedMethod();
+            }
+
+         // Ugly; this will be cleaned up in a future PR
```

Comment:
looking forward to that day :)

Reply:
May be a good idea to open an issue and reference this review comment. These sorts of things tend to easily get lost our huge codebase if you're not tracked.

Reply:
https://github.com/eclipse/openj9/issues/3028 is basically the issue that would result in this cleaned. Either that or https://github.com/eclipse/openj9/issues/3031 in which we add a new issue to fix the relo headers from being of size UDATA and/or allowing unions within the struct; the latter is blocked by the issue this PR is partially addressing (https://github.com/eclipse/openj9/issues/4803).